internal/oc/release_test.go

```diff
@@ -259,29 +263,88 @@ var _ = Describe("oc", func() {
+				args := splitStringToInterfacesArray(command)
+				imageInfoStr := fmt.Sprintf("{ \"config\": { \"architecture\": \"%s\" }}", common.TestDefaultConfig.CPUArchitecture)
+				mockExecuter.EXPECT().Execute(args[0], args[1:]...).Return(imageInfoStr, "", 0).Times(1)
 
-			arch, err := oc.GetReleaseArchitecture(log, releaseImage, "", pullSecret)
-			Expect(arch).Should(Equal(common.TestDefaultConfig.CPUArchitecture))
-			Expect(err).ShouldNot(HaveOccurred())
+				arch, err := oc.GetReleaseArchitecture(log, releaseImage, "", pullSecret)
+				Expect(arch).Should(Equal([]string{common.TestDefaultConfig.CPUArchitecture}))
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			It("fail with malformed cpu architecture", func() {
+				command := fmt.Sprintf(templateImageInfo+" --registry-config=%s", releaseImage, tempFilePath)
+				args := splitStringToInterfacesArray(command)
+				imageInfoStr := fmt.Sprintf("{ \"config\": { \"not-an-architecture\": \"%s\" }}", common.TestDefaultConfig.CPUArchitecture)
+				mockExecuter.EXPECT().Execute(args[0], args[1:]...).Return(imageInfoStr, "", 0).Times(1)
+
+				arch, err := oc.GetReleaseArchitecture(log, releaseImage, "", pullSecret)
+				Expect(arch).Should(BeEmpty())
+				Expect(err).Should(HaveOccurred())
+			})
 		})
 
-		It("fetch cpu architecture from multiarch release image", func() {
-			command := fmt.Sprintf(templateImageInfo+" --registry-config=%s", releaseImage, tempFilePath)
-			command2 := fmt.Sprintf(templateSkopeoDetectMultiarch+" --authfile %s", releaseImage, tempFilePath)
-			args := splitStringToInterfacesArray(command)
-			args2 := splitStringToInterfacesArray(command2)
-			imageInfoStr := fmt.Sprintf("{ \"manifests\": { \"platform\": { \"architecture\": \"%s\" }}}", common.TestDefaultConfig.CPUArchitecture)
-			mockExecuter.EXPECT().Execute(args[0], args[1:]...).Return("", "the image is a manifest list", 1).Times(1)
-			mockExecuter.EXPECT().Execute(args2[0], args2[1:]...).Return(imageInfoStr, "", 0).Times(1)
+		Context("for multi-arch release image", func() {
+			It("fetch cpu architecture", func() {
+				command := fmt.Sprintf(templateImageInfo+" --registry-config=%s", releaseImage, tempFilePath)
+				command2 := fmt.Sprintf(templateSkopeoDetectMultiarch+" --authfile %s", releaseImage, tempFilePath)
+				args := splitStringToInterfacesArray(command)
+				args2 := splitStringToInterfacesArray(command2)
+				mockExecuter.EXPECT().Execute(args[0], args[1:]...).Return("", "the image is a manifest list", 1).Times(1)
+				mockExecuter.EXPECT().Execute(args2[0], args2[1:]...).Return(test_skopeo_multiarch_image_output, "", 0).Times(1)
 
-			arch, err := oc.GetReleaseArchitecture(log, releaseImage, "", pullSecret)
-			Expect(arch).Should(Equal(common.MultiCPUArchitecture))
-			Expect(err).ShouldNot(HaveOccurred())
+				arch, err := oc.GetReleaseArchitecture(log, releaseImage, "", pullSecret)
+				Expect(arch).Should(Equal([]string{"x86_64", "ppc64le", "s390x", "arm64"}))
```

Comment:
Does this check validate the order? If so, is there a matcher that doesn't?

Reply:
Yes it does, but it runs against a manually crafted `test_skopeo_multiarch_image_output` which has the same order. Because of that, as long as we don't change our implementation of unmarshalling that would stop preserving the order, this is a "safe" check.

Does it answer or did I get the question wrong?

Reply:
Ah I looked it up, looks like there's a matcher for what I'm talking about.

`ConsistOf` https://onsi.github.io/gomega/#consistofelement-interface

Really my suggestion is that no matter what happens with the implementation as long as the elements are the same the test should still pass. Using a matcher like this just makes the test a bit less fragile.

Reply:
Done